Fix create_synonyms_set not working

pmusa opened the pull request:
According to the [documentation](https://www.elastic.co/guide/en/app-search/current/synonyms.html#synonyms-create), `create_synonyms_set` should send a body containing the synonyms list. Currently, the code is sending it as a URL parameter, which fails. This PR, fixes the issue by sending the synonyms list through the body.

JasonStoltz commented:
@yakhinvadim Please have a look, I've made the changes you mentioned.

yakhinvadim commented on elastic_enterprise_search/apis/documents.py:
```diff
@@ -0,0 +1,74 @@
+#  https://swiftype.com/documentation/enterprise-search/api/custom-sources
+class Documents:
+
+    def __init__(self, session):
```
[Here](https://github.com/elastic/enterprise-search-python/pull/19/files?file-filters%5B%5D=.py#diff-778c38965813f7e6e5c5d009ec364810R15) we're passing `(self.session)` as a first and single argument.

But in this line we accept two arguments: `self` and `session`, and `session` is the second.

Is the first argument of a constructor __init__ reserved for `self` here and everything's ok?

(not to myself: question about next line)

JasonStoltz replied:
Yes, as far I understand it the first parameter of any Class method is a reference to the instance, and calling it `self` is just a standard convention. 

https://docs.python.org/3/tutorial/classes.html#random-remarks

Discussed here: https://docs.python.org/3/tutorial/classes.html#random-remarks

JasonStoltz commented on README.md:
```diff
@@ -53,7 +53,7 @@ Indexing a document into a custom content source:
     }
   ]
 
-  document_results = client.index_documents(content_source_key, documents, timeout=10, delay=2)
```
Correct. That's from when there was an async version of indexing.

pmusa committed:
Fix create_synonyms_set not working

According to the [documentation](https://www.elastic.co/guide/en/app-search/current/synonyms.html#synonyms-create), `create_synonyms_set` should send a body containing the synonyms list. Currently, the code is sending it as a URL parameter, which fails. This PR, fixes the issue by sending the synonyms list through the body.

sethmlarson commented on elastic_enterprise_search/client/_app_search.py:
```diff
@@ -1395,12 +1395,6 @@ def create_synonym_set(
             if param in SKIP_IN_PATH:
                 raise ValueError("Empty value passed for a required argument")
 
-        params = make_params(
```
Everything under `client/` is auto-generated from specs, so if this is failing then this is likely a specification problem. I'll investigate.

sethmlarson commented:
Closing this in favor of updating the specification directly. Will regenerate once that lands.
